spec/text-document-spec.coffee

```diff
@@ -2,28 +2,51 @@ Point = require "../src/point"
 TextDocument = require "../src/text-document"
 
 describe "TextDocument", ->
-  document = null
-
-  beforeEach ->
-    document = new TextDocument
+  [document, layer] = []
 
   describe "::buildDisplayLayer()", ->
     describe "::slice(start, end)", ->
       it "returns the content between the start and end points", ->
+        document = new TextDocument(softWrapColumn: 10, tabLength: 4)
```

Comment:
Editors will need to construct their own `TextDisplayDocument` (which will wrap the display layers) because they could have different parameters such as soft wrap information, tab length, etc. Also, each `TextDisplayDocument` will contain its own set of folds.